async-nats/src/jetstream/mod.rs

```diff
@@ -198,3 +198,27 @@ pub fn with_domain<T: AsRef<str>>(client: Client, domain: T) -> Context {
 pub fn with_prefix(client: Client, prefix: &str) -> Context {
     context::Context::with_prefix(client, prefix)
 }
+
+/// Checks if a name passed in JetStream API is valid one.
+/// The restrictions are there because some fields in the JetStream configs are passed as part of the subject to apply permissions.
+/// Examples are stream names, consumer names, etc.
+pub(crate) fn is_valid_name(name: &str) -> bool {
+    !name.is_empty()
+        && !name.contains(|c| char::is_ascii_whitespace(&c) || c == '.')
+        && name != ">"
+        && name != "*"
```

Comment:
Going by Compiler Explorer this should be much faster https://rust.godbolt.org/z/6cTGrhqar

```suggestion
    !matches!(name, "" | ">" | "*") && name.bytes().all(|c| !c.is_ascii_whitespace() && c != b'.')
```

The second part iterates on the string as a slice of bytes, instead of an  `Iterator` of `char`s. This optimization is possible only because we're interested in matching ASCII characters.

Reply:
I just realized that although it is fine to have wildcards in subject, Server rejects those anyway, so will make them as part of `contains`

Reply:
The code is even smaller that way :smile: https://rust.godbolt.org/z/5o86Ex648